### ISSUE TITLE
Host custom configs server-side?

### ISSUE THREAD

**Issue by bagage** [nrenner/brouter-web#97]
On brouter.damsy.net I'd like to add some profiles (mainly MTB and trekking-no-path). 
But since I use brouter engine from brouter.de, it seems I must also use the profiles located on it (eg http://brouter.de/brouter/profiles2/).

@nrenner is there any way to host some "custom" profiles server-side?

**Comment by poutnikl** [comment]
Including custom profiles to the brouter-web menu offering is definitely possible in local brouter-web installation. That was the main reason for my experimenting with local brouter-web server. 

The config.js file of the brrouter-web folder explicitly enumerates the profiles for the menu. The profile presence in profiles2 subfolder is not enough. Al least this was true about year ago. 

I guess if such a modified config.file is provided among the server files..........

**Comment by nrenner** [comment]
> But since I use brouter engine from brouter.de, it seems I must also use the profiles located on it (eg http://brouter.de/brouter/profiles2/).

Yes, the engine can only load profiles from that directory and uploaded custom session profiles.

> @nrenner is there any way to host some "custom" profiles server-side?

Currently not. There is the idea to be able to make uploaded custom profiles more permanent for sharing and bookmarking (and potentially adding to local dropdown), see related issues #9 and #26. But that would not change the publicly available profile list.

As @poutnikl said, the current solution/workaround for having a custom set of profiles is to install your own server instance.

**Comment by bagage** [comment]
OK thanks to both :). I'm currently developing an extension to brouter-web to support extra profiles - more to come later.

**Comment by bagage** [comment]
For information I started some WIP in my branch: https://github.com/bagage/brouter-web/commits/dev 
Basically when user selects a custom server profile, brouter-web automatically uploads it and switch to custom profile. It's not an ideal solution but at least it requires minor changes to be effective and up on demo server brouter.damsy.net
Among bugs and improvements:
- an error popup appears temporary between selection of custom server profile and the time of profile is uploaded on server. This popup should not appears.
- instead of changing profile to "custom" we should stick to the profile selected instead, since user did not mean to use a custom profile…

**Comment by nrenner** [comment]
Thanks for the info. Glad you found a solution.

**Comment by AntoineLu** [comment]
Best would actually be to be able to upload a profile from a URL. This way, we could store our profile on our personal github for example and then just upload it from a "raw" url to the file.
